File: Npc/11004048.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004048: Carriage Tender
/// </summary>
public class _11004048 : NpcScript {
    protected override int First() {
        return 10;
    }

    // Select 0:
    // $script:0614185307010069$
    // - It's a long walk to the Frontier Foundation. Why not take a carriage?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (10, 0):
                // $script:0614185307010070$
                // - It's a long walk to the Frontier Foundation. Why not take a carriage?
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (10, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11004048.py
""" 11004048: Carriage Tender """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0614185307010069$
        # - It's a long walk to the Frontier Foundation. Why not take a carriage?
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0614185307010070$
        # - It's a long walk to the Frontier Foundation. Why not take a carriage?
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE
